src/kv/tx.h

```diff
@@ -74,13 +80,101 @@ namespace kv
       return std::make_tuple(typed_view);
     }
 
+    template <class M>
+    std::tuple<typename M::TxView*> get_tuple2(const std::string& map_name)
+    {
+      if (store == nullptr)
+      {
+        throw std::logic_error("New form called on old-style Tx");
+      }
+
+      using MapView = typename M::TxView;
+
+      // If the M is present, its AbstractTxView should be an M::TxView. This
+      // invariant could be broken by set_view_list, which will produce an error
+      // here
+      auto search = view_list.find(map_name);
+      if (search != view_list.end())
+      {
+        auto view = dynamic_cast<MapView*>(search->second.view.get());
+
+        if (view == nullptr)
+        {
+          throw std::logic_error(
+            fmt::format("View over map {} is not of expected type", map_name));
+        }
+
+        return std::make_tuple(view);
+      }
+
+      if (read_version == NoVersion)
+      {
+        // Grab opacity version that all Maps should be queried at.
+        auto txid = store->current_txid();
+        term = txid.term;
+        read_version = txid.version;
+      }
+
+      M* typed_map = nullptr;
+
+      auto type_erased_map = store->get_map(read_version, map_name);
+      if (type_erased_map == nullptr)
+      {
+        // Store doesn't know this map yet - create it dynamically
+        {
+          const auto map_it = created_maps.find(map_name);
+          if (map_it != created_maps.end())
+          {
+            throw std::logic_error("Created map without creating view over it");
```

Comment:
Does this make `tx.get_view2()` non-idempotent? My understanding is that it would be preferable for it to be idempotent?

Reply:
Ah! But what about if the type of the map is different between the two calls to `get_view2()`? This should definitely fail in that case.

Reply:
I don't understand the idempotence concern. This check is really just dumb and should never be true, but if it is (as with the error for non-`AbstractTxView` below) we want a loud failure to tell us some expectation is not true.

Reply:
As discussed, this is fine because if we call `get_view2` twice with the same `"map"` on the same `tx`, the `map` would be in the `view_list` on the second call and we would return early.